platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -485,6 +485,25 @@ static void store_sa_info(ipsec_sa_t *ipsec_sa, const odp_ipsec_sa_param_t *p)
 		ipsec_sa->sa_info.out.mtu = p->outbound.mtu;
 }
 
+static int init_cbc_salt(ipsec_sa_t *ipsec_sa)
+{
+	int filled = 0;
+	int rc;
+
+	if (!ipsec_sa->use_cbc_iv)
```

Comment:
Matter of taste but as the function is called `init_cbc_salt()` it should should just init it unconditionally and this could be moved outside.

Reply:
I prefer it this way and do not feel that the naming strongly indicates that the checking could not be in the init function (i.e. now the init function inits what is necessary). Maybe it is not exactly equivalent but odp_ipsec_sad_init_global() is also called unconditionally and internally checks if anything actually needs to be inited.

Reply:
The point here was that data for the condition is known (all internal, within the same module), as opposed to e.g. the `_odp_ipsec_sad_init_global()` which is an "internal API" call, called from different modules that don't know the specifics. But like I said, matter of taste.